src/api/admin/store/route.ts

```diff
@@ -0,0 +1,46 @@
+import { MedusaRequest, MedusaResponse } from "@medusajs/medusa";
+import { EntityManager } from "typeorm";
+import StoreService from "../../../services/store";
+import { Store } from "../../../models/store";
+
+/**
+ * [get] /store
+ * description: get current user store
+ */
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const manager: EntityManager = req.scope.resolve("manager");
+  const storeService: StoreService = req.scope.resolve("storeService");
+
+  const store = await manager.transaction(async (transactioManager) => {
+    const storeServiceWithTransaction =
+      storeService.withTransaction(transactioManager);
+
+    return await storeServiceWithTransaction.retrieve();
+  });
+
+  res.status(200).json({
+    store,
+  });
+};
+
+/**
+ * [post] /store
+ * creates store with payload
+ */
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
```

Comment:
note: by default POST request in Medusa for Store is tailored just to update the store of the logged user
since we aim to create store entries from CT sellers/vendors, we need to establish a mechanism to create a store with given payload.
this function handles this need